File: code/mathematical-algorithms/babylonian_method/babylonian_method.go
package main

import "fmt"

func squareRoot(n int) float64 {
	x := float64(n)
	nF := float64(n)
	e := 0.000001

	for x - (nF/x) > e {
	    x = ((nF/x)+x)/2
	}

	return x
}


func main() {
	a := 90
    fmt.Printf("The square root of %v is %v \n", a, squareRoot(a))
}

File: code/mathematical_algorithms/babylonian_method/babylonian_method.java
import java.util.*;
class Babylonian{
    public static void main(String args[] ) throws Exception {
        Scanner s = new Scanner(System.in);
        double n = s.nextDouble();
        System.out.println("The square root of " + n + " is " + squareRoot(n));
    }

    /*Returns the square root of n*/
    public static double squareRoot(double n)
    {
      double x = n;
      double y = 1;
      double acc = 0.000001; /* acc decides the accuracy level*/
      while(x - y > acc)
      {
        x = (x + y)/2;
        y = n/x;
      }
      return x;
    }
}